sched/task/task_start.c

```diff
@@ -134,18 +134,18 @@ void nxtask_start(void)
    * we have to switch to user-mode before calling the task.
    */
 
-#ifndef CONFIG_BUILD_FLAT
   if ((tcb->cmn.flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL)
     {
+#ifndef CONFIG_BUILD_FLAT
```

Comment:
How about making this positive logic?

Reply:
Done.

Reply:
@xiaoxiang781216 What about @jerpelea concerns?